tools/gfx/slang-context.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "tools/gfx/render.h"
+
+namespace gfx
+{
+    class SlangContext
+    {
+    public:
+        Slang::ComPtr<slang::IGlobalSession> globalSession;
+        Slang::ComPtr<slang::ISession> session;
+        Result initialize(const gfx::IRenderer::SlangDesc& desc, SlangCompileTarget compileTarget, const char* defaultProfileName)
+        {
+            if (desc.slangGlobalSession)
+            {
+                globalSession = desc.slangGlobalSession;
+            }
+            else
+            {
+                SLANG_RETURN_ON_FAIL(slang::createGlobalSession(globalSession.writeRef()));
+            }
+
+            slang::SessionDesc slangSessionDesc = {};
+            slangSessionDesc.defaultMatrixLayoutMode = desc.defaultMatrixLayoutMode;
+            slangSessionDesc.searchPathCount = desc.searchPathCount;
+            slangSessionDesc.searchPaths = desc.searchPaths;
+            slangSessionDesc.preprocessorMacroCount = desc.preprocessorMacroCount;
+            slangSessionDesc.preprocessorMacros = desc.preprocessorMacros;
+            slang::TargetDesc targetDesc = {};
+            targetDesc.format = compileTarget;
+            auto targetProfile = desc.targetProfile;
+            if (targetProfile == nullptr)
+                targetProfile = defaultProfileName;
+            targetDesc.profile = globalSession->findProfile(targetProfile);
```

Comment:
I guess this is one detail that makes it tricky to just use `SessionDesc`: that type requires the integer version of the profile, while we want users of this API to be able to use the string.

That gets into a messier issue which is that we probably need to make one of two choices in the Slang API:

* We could support/encourage using strings directly to represent profiles, rather than having to go through `spFindProfile()` or `IGlobalSession->findProfile()`. The cost of the string lookups is realistically neglibible at this point compared to other work being done inside the Slang library.

* We could decide to give most (or all) profiles fixed integer ID values so that application code is not required to look them up by string name at runtime. Instead, they could hard-code in a value using an `enum` defined in `slang.h`.

I'm not sure which of these we should prefer, but it seems like either one would be part of the way to get around mess like this.